exporters/filewriter/file_exporter.go

```diff
@@ -0,0 +1,135 @@
+package filewriter
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/plugins"
+	"github.com/sirupsen/logrus"
+)
+
+const exporterName = "filewriter"
+
+type fileExporter struct {
+	round  uint64
+	fd     *os.File
+	cfg    ExporterConfig
+	logger *logrus.Logger
+}
+
+var fileExporterMetadata = exporters.ExporterMetadata{
+	ExpName:        exporterName,
+	ExpDescription: "Exporter for writing data to a file.",
+	ExpDeprecated:  false,
+}
+
+// Constructor is the ExporterConstructor implementation for the filewriter exporter
+type Constructor struct{}
+
+// New initializes a fileExporter
+func (c *Constructor) New() exporters.Exporter {
+	return &fileExporter{
+		round: 0,
+	}
+}
+
+func (exp *fileExporter) Metadata() exporters.ExporterMetadata {
+	return fileExporterMetadata
+}
+
+func (exp *fileExporter) Init(cfg plugins.PluginConfig, logger *logrus.Logger) error {
```

Comment:
can PluginConfig be a file path to a configs file? For the file exporter, Close() should update the round in the configs file so the value doesn't need to be manually updated.

Reply:
> can PluginConfig be a file path to a configs file?

I'm not sure what you mean. The `PluginConfig` is just a wrapper around a string. Each plugin then deserializes that into expected values via defined configs such as the one in `file_exporter_config.go`.

> For the file exporter, Close() should update the round in the configs file so this value doesn't need to be manually updated.

Yes, this sounds correct--that sort of thing is exactly why we have the `Close` function.